sycl/plugins/level_zero/pi_level_zero.hpp

```diff
@@ -723,11 +727,26 @@ struct _pi_queue : _pi_object {
     // Level Zero command queue handles.
     std::vector<ze_command_queue_handle_t> ZeQueues;
 
+    // Immediate commandlist handles, one per Level Zero command queue handle.
+    // These are created only once, along with the L0 queues (see above)
+    // and reused thereafter.
+    std::vector<pi_command_list_ptr_t> ImmCmdLists;
+
+    // Return the index of the next queue to use based on a
+    // round robin strategy and the queue group ordinal.
+    uint32_t getQueueIndex(uint32_t *QueueGroupOrdinal, uint32_t *QueueIndex);
+
+    // Return a queue descriptor using the provided ordinal and index.
+    void createQueueDesc(uint32_t Index, uint32_t Ordinal,
+                         ZeStruct<ze_command_queue_desc_t> &ZeCommandQueueDesc);
+
     // This function will return one of possibly multiple available native
-    // queues. Currently, a round robin strategy is used. This function also
-    // sends back the value of the queue group ordinal.
+    // queues and the value of the queue group ordinal.
     ze_command_queue_handle_t &getZeQueue(uint32_t *QueueGroupOrdinal);
```

Comment:
this one could be changed to return the queue descriptor as well, and not just the ordinal